commit 8ea4964d2af55b62bc87aba54b1db58c7c604a1c
Author: Slaven Rezic <slaven@rezic.de>
Date:   Tue Feb 2 14:56:32 2010 +0100

    * MMtry improvements
      * VERBOSE is now even more verbose by showing all of compiler stderr
        output
      * use $Config{ldflags} when trying to compile things

diff --git a/Tk/MMtry.pm b/Tk/MMtry.pm
index 110f6b2..58acd13 100644
--- a/Tk/MMtry.pm
+++ b/Tk/MMtry.pm
@@ -23,9 +23,10 @@ sub try_compile
  $inc = [] unless $inc;
  $lib = [] unless $lib;
  $def = [] unless $def;
+ my $stderr_too = $VERBOSE ? '' : $stderr_too;
  my $out   = basename($file,'.c').$Config{'exe_ext'};
  warn "Test Compiling $file\n";
- my $msgs  = `$Config{'cc'} -o $out $Config{'ccflags'} @$inc $file @$lib @$def $stderr_too`;
+ my $msgs  = `$Config{'cc'} -o $out $Config{'ccflags'} @$inc $file $Config{ldflags} @$lib @$def $stderr_too`;
  my $ok = ($? == 0);
  warn "$msgs\n" if $VERBOSE && $msgs;
  unlink($out) if (-f $out);
@@ -38,9 +39,10 @@ sub try_run
  $inc = [] unless $inc;
  $lib = [] unless $lib;
  $def = [] unless $def;
+ my $stderr_too = $VERBOSE ? '' : $stderr_too;
  my $out   = basename($file,'.c').$Config{'exe_ext'};
  warn "Test Compile/Run $file\n";
- my $cmdline = "$Config{'cc'} -o $out $Config{'ccflags'} @$inc $file @$lib @$def";
+ my $cmdline = "$Config{'cc'} -o $out $Config{'ccflags'} @$inc $file $Config{ldflags} @$lib @$def";
  my $msgs  = `$cmdline $stderr_too`;
  my $ok = ($? == 0);
  warn "$cmdline:\n$msgs\n" if $VERBOSE && $msgs;
